Compilation fails on the most recent stable rustc

## Environment

Rust: 1.46.0
OS: Ubuntu 20.04 under WSL 2.

## Issue

I tried to run the tests as described in [development guide](https://github.com/popzxc/RustPython/blob/master/DEVELOPMENT.md#testing) (basically, installed dependencies and ran `cargo test --all`).

However, compilation failed with the following errors:

```sh
error[E0277]: `*mut u8` cannot be shared between threads safely
   --> wasm/lib/src/browser_module.rs:165:6
    |
165 | impl PyValue for PyPromise {
    |      ^^^^^^^ `*mut u8` cannot be shared between threads safely
    |
    = help: within `browser_module::PyPromise`, the trait `std::marker::Sync` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `js_sys::Promise`
    = note: required because it appears within the type `browser_module::PyPromise`

error[E0277]: `*mut u8` cannot be sent between threads safely
   --> wasm/lib/src/browser_module.rs:165:6
    |
165 | impl PyValue for PyPromise {
    |      ^^^^^^^ `*mut u8` cannot be sent between threads safely
    |
    = help: within `browser_module::PyPromise`, the trait `std::marker::Send` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `js_sys::Promise`
    = note: required because it appears within the type `browser_module::PyPromise`

error[E0277]: `*mut u8` cannot be shared between threads safely
   --> wasm/lib/src/browser_module.rs:261:6
    |
261 | impl PyValue for Document {
    |      ^^^^^^^ `*mut u8` cannot be shared between threads safely
    |
    = help: within `browser_module::Document`, the trait `std::marker::Sync` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `web_sys::EventTarget`
    = note: required because it appears within the type `web_sys::Node`
    = note: required because it appears within the type `web_sys::Document`
    = note: required because it appears within the type `browser_module::Document`

error[E0277]: `*mut u8` cannot be sent between threads safely
   --> wasm/lib/src/browser_module.rs:261:6
    |
261 | impl PyValue for Document {
    |      ^^^^^^^ `*mut u8` cannot be sent between threads safely
    |
    = help: within `browser_module::Document`, the trait `std::marker::Send` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `web_sys::EventTarget`
    = note: required because it appears within the type `web_sys::Node`
    = note: required because it appears within the type `web_sys::Document`
    = note: required because it appears within the type `browser_module::Document`

error[E0277]: `*mut u8` cannot be shared between threads safely
   --> wasm/lib/src/browser_module.rs:287:6
    |
287 | impl PyValue for Element {
    |      ^^^^^^^ `*mut u8` cannot be shared between threads safely
    |
    = help: within `browser_module::Element`, the trait `std::marker::Sync` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `web_sys::EventTarget`
    = note: required because it appears within the type `web_sys::Node`
    = note: required because it appears within the type `web_sys::Element`
    = note: required because it appears within the type `browser_module::Element`

error[E0277]: `*mut u8` cannot be sent between threads safely
   --> wasm/lib/src/browser_module.rs:287:6
    |
287 | impl PyValue for Element {
    |      ^^^^^^^ `*mut u8` cannot be sent between threads safely
    |
    = help: within `browser_module::Element`, the trait `std::marker::Send` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `web_sys::EventTarget`
    = note: required because it appears within the type `web_sys::Node`
    = note: required because it appears within the type `web_sys::Element`
    = note: required because it appears within the type `browser_module::Element`

error: aborting due to 6 previous errors

For more information about this error, try `rustc --explain E0277`.
error: could not compile `rustpython_wasm`.

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error[E0277]: `*mut u8` cannot be shared between threads safely
   --> wasm/lib/src/browser_module.rs:165:6
    |
165 | impl PyValue for PyPromise {
    |      ^^^^^^^ `*mut u8` cannot be shared between threads safely
    |
    = help: within `browser_module::PyPromise`, the trait `std::marker::Sync` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `js_sys::Promise`
    = note: required because it appears within the type `browser_module::PyPromise`

error[E0277]: `*mut u8` cannot be sent between threads safely
   --> wasm/lib/src/browser_module.rs:165:6
    |
165 | impl PyValue for PyPromise {
    |      ^^^^^^^ `*mut u8` cannot be sent between threads safely
    |
    = help: within `browser_module::PyPromise`, the trait `std::marker::Send` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `js_sys::Promise`
    = note: required because it appears within the type `browser_module::PyPromise`

error[E0277]: `*mut u8` cannot be shared between threads safely
   --> wasm/lib/src/browser_module.rs:261:6
    |
261 | impl PyValue for Document {
    |      ^^^^^^^ `*mut u8` cannot be shared between threads safely
    |
    = help: within `browser_module::Document`, the trait `std::marker::Sync` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `web_sys::EventTarget`
    = note: required because it appears within the type `web_sys::Node`
    = note: required because it appears within the type `web_sys::Document`
    = note: required because it appears within the type `browser_module::Document`

error[E0277]: `*mut u8` cannot be sent between threads safely
   --> wasm/lib/src/browser_module.rs:261:6
    |
261 | impl PyValue for Document {
    |      ^^^^^^^ `*mut u8` cannot be sent between threads safely
    |
    = help: within `browser_module::Document`, the trait `std::marker::Send` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `web_sys::EventTarget`
    = note: required because it appears within the type `web_sys::Node`
    = note: required because it appears within the type `web_sys::Document`
    = note: required because it appears within the type `browser_module::Document`

error[E0277]: `*mut u8` cannot be shared between threads safely
   --> wasm/lib/src/browser_module.rs:287:6
    |
287 | impl PyValue for Element {
    |      ^^^^^^^ `*mut u8` cannot be shared between threads safely
    |
    = help: within `browser_module::Element`, the trait `std::marker::Sync` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `web_sys::EventTarget`
    = note: required because it appears within the type `web_sys::Node`
    = note: required because it appears within the type `web_sys::Element`
    = note: required because it appears within the type `browser_module::Element`

error[E0277]: `*mut u8` cannot be sent between threads safely
   --> wasm/lib/src/browser_module.rs:287:6
    |
287 | impl PyValue for Element {
    |      ^^^^^^^ `*mut u8` cannot be sent between threads safely
    |
    = help: within `browser_module::Element`, the trait `std::marker::Send` is not implemented for `*mut u8`
    = note: required because it appears within the type `std::marker::PhantomData<*mut u8>`
    = note: required because it appears within the type `wasm_bindgen::JsValue`
    = note: required because it appears within the type `js_sys::Object`
    = note: required because it appears within the type `web_sys::EventTarget`
    = note: required because it appears within the type `web_sys::Node`
    = note: required because it appears within the type `web_sys::Element`
    = note: required because it appears within the type `browser_module::Element`

error: aborting due to 6 previous errors
```

```
/Users/me/.cargo/bin/cargo build --color=always --all --all-targets
   Compiling rustpython_wasm v0.1.2 (/opt/RustPython/wasm/lib)
error[E0277]: `*mut u8` cannot be shared between threads safely
   --> wasm/lib/src/browser_module.rs:162:6
    |
162 | impl PyValue for Document {
    |      ^^^^^^^ `*mut u8` cannot be shared between threads safely
```

any ways to mitigate?

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -71,7 +71,7 @@ Python unit tests can be run by compiling RustPython and running the test module
 $ cargo run --release -- -m test
 ```
 
-There are a few test options that are specially useful:
+There are a few test options that are especially useful:
 
 - `-j <n>` enables parallel testing (which is a lot faster), where `<n>` is the
 number of threads to be used, ideally the same as number of cores on your CPU.

--- a/DEVELOPMENT.md
+++ b/DEVELOPMENT.md
@@ -62,7 +62,34 @@ $ pytest -v
 Rust unit tests can be run with `cargo`:
 
 ```shell
-$ cargo test --all
+$ cargo test --workspace --exclude rustpython_wasm
+```
+
+Python unit tests can be run by compiling RustPython and running the test module:
+
+```shell
+$ cargo run --release -- -m test
+```
+
+There are a few test options that are specially useful:
+
+- `-j <n>` enables parallel testing (which is a lot faster), where `<n>` is the
+number of threads to be used, ideally the same as number of cores on your CPU.
+If you don't know, `-j 4` or `-j 8` are good options.
+- `-v` enables verbose mode, adding additional information about the tests being
+run.
+- `<test_name>` specifies a single test to run instead of running all tests.
+
+For example, to run all tests in parallel:
+
+```shell
+$ cargo run --release -- -m test -j 4
+```
+
+To run only `test_cmath` (located at `Lib/test/test_cmath`) verbosely:
+
+```shell
+$ cargo run --release -- -m test test_cmath -v
 ```
 
 ## Profiling

